src/main/java/linenux/command/ViewCommand.java

```diff
@@ -0,0 +1,159 @@
+    private static final String NUMBER_PATTERN = "^\\d+$";
+    private static final String CANCEL_PATTERN = "^cancel$";
+
+    private Schedule schedule;
+    private boolean requiresUserResponse;
+    private ArrayList<Task> foundTasks;
+
+    public ViewCommand(Schedule schedule) {
+        this.schedule = schedule;
+        this.requiresUserResponse = false;
+    }
+
+    @Override
+    public String getTriggerWord() {
+        return TRIGGER_WORD;
+    }
+
+    @Override
+    public String getDescription() {
+        return DESCRIPTION;
+    }
+
+    @Override
+    public boolean respondTo(String userInput) {
+        return userInput.matches(VIEW_PATTERN);
+    }
+
+    @Override
+    public CommandResult execute(String userInput) {
+        Matcher matcher = Pattern.compile(VIEW_PATTERN).matcher(userInput);
+
+        if (matcher.matches()) {
+            assert (this.schedule != null);
+
+            String keywords = matcher.group("keywords");
+            String[] keywordsArr = keywords.split("\\s+");
+            ArrayList<Task> tasks = this.schedule.search(keywordsArr);
+
+            if (tasks.size() == 0) {
+                return makeNotFoundResult(keywords);
+            } else if (tasks.size() == 1) {
+                return makeResult(tasks.get(0));
+            } else {
+                this.requiresUserResponse = true;
+                this.foundTasks = tasks;
+                return makePromptResult(tasks);
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public boolean awaitingUserResponse() {
+        return requiresUserResponse;
+    }
+
+    @Override
+    public CommandResult userResponse(String userInput) {
+        if (userInput.matches(NUMBER_PATTERN)) {
+            int index = Integer.parseInt(userInput);
+
+            if (1 <= index && index <= this.foundTasks.size()) {
+                Task task = this.foundTasks.get(index - 1);
+
+                this.requiresUserResponse = false;
+                this.foundTasks = null;
+                return makeResult(task);
+            } else {
+                return makeInvalidIndexResult();
+            }
+        } else if (userInput.matches(CANCEL_PATTERN)) {
+            this.requiresUserResponse = false;
+            this.foundTasks = null;
+            return makeCancelledResult();
+        } else {
+            return makeInvalidUserResponse(userInput);
+        }
+    }
+
+    private CommandResult makeNotFoundResult(String keywords) {
+        return () -> "Cannot find \"" + keywords + "\".";
+    }
+
+    private CommandResult makeResult(Task task) {
+        ArrayList<Reminder> reminders = task.getReminders();
+        StringBuilder builder = new StringBuilder();
+        builder.append(task.toString());
+        builder.append('\n');
+        builder.append("Reminders:" + '\n');
+
+        if (reminders.size() == 0) {
+            builder.append("There are no reminders found!");

```

Comment:
Hmm, I think I should've rephrased it. I was going along the lines of "You have not set any reminders for this task." to show the user that it's empty and not missing.